packages/edit-post/src/editor.js

```diff
@@ -24,6 +26,14 @@ import { unlock } from './lock-unlock';
 import useNavigateToEntityRecord from './hooks/use-navigate-to-entity-record';
 
 const { ExperimentalEditorProvider } = unlock( editorPrivateApis );
+const { BlockRemovalWarningModal } = unlock( blockEditorPrivateApis );
+// Prevent accidental removal of certain blocks, asking the user for
+// confirmation.
+const blockRemovalRules = {
+	'bindings/core/pattern-overrides': __(
+		'Blocks from synced patterns that can have overriden content.'
```

Comment:
Are these messages that are specific per block type? Should this be a block API instead?

Reply:
I don't think they are, pattern overrides could in the future potentially be on any block, so it wouldn't make sense to add the rule to every individual block. Unless it was a default part of blocks that support it (it's undecided right now how determining support will work, as it's dictated by the blocks that support bindings), but I think even then that might be a bit much.

I think it could be better as something that can be declared more centrally, but this API does need to be different 
to where it landed here IMO.